api/v1alpha1/k8ssandracluster_types.go

```diff
@@ -76,42 +76,77 @@ type K8ssandraClusterList struct {
 	Items           []K8ssandraCluster `json:"items"`
 }
 
-type Cassandra struct {
+type CassandraClusterTemplate struct {
+	// Cluster is the name of the cluster. This corresponds to cluster_name in
+	// cassandra.yaml.
+	// +kubebuilder:validation:MinLength=2
 	Cluster string `json:"cluster,omitempty"`
 
-	Datacenters []CassandraDatacenterTemplateSpec `json:"datacenters,omitempty"`
+	// ServerImage is the image for the cassandra container. Note that this should be a
+	// management-api image. If left empty the operator will choose a default image based
+	// on ServerVersion.
+	ServerImage string `json:"serverImage,omitempty"`
+
+	// ServerVersion is the Cassandra version.
+	// +kubebuilder:validation:Pattern=(3\.11\.\d+)|(4\.0\.\d+)
+	ServerVersion string `json:"serverVersion,omitempty"`
+
+	// Resources is the cpu and memory resources for the cassandra container.
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// SystemLoggerResources is the cpu and memory resources for the server-system-logger
+	// container.
+	SystemLoggerResources *corev1.ResourceRequirements `json:"systemLoggerResources,omitempty"`
+
+	// CassandraConfig is configuration settings that are applied to cassandra.yaml and
+	// jvm-options for 3.11.x or jvm-server-options for 4.x.
+	CassandraConfig *CassandraConfig `json:"config,omitempty"`
+
+	// StorageConfig is the persistent storage requirements for each Cassandra pod. This
+	// includes everything under /var/lib/cassandra, namely the commit log and data
+	// directories.
+	StorageConfig *cassdcapi.StorageConfig `json:"storageConfig,omitempty"`
```

Comment:
If StorageConfig is left out (omit empty), what's the intended behaviour?

Reply:
Validation is needed to ensure that it is set in either the CassandraClusterTemplate or the CassandraDatacenterTemplate (or both).

Reply:
I created #121 to address validation checks.